src/bun.js/bindings/BunObject.cpp

```diff
@@ -362,6 +362,8 @@ JSC_DEFINE_HOST_FUNCTION(functionBunSleep,
         return JSC::JSValue::encode(JSC::JSValue {});
     }
 
+    millisecondsValue = JSC::jsNumber(std::ceil(millisecondsValue.asNumber()));
```

Comment:
this shouldn't have to re-encode a JSValue. though `jsNumber` isn't expensive it is needless as the value will be immediatly converted back to an i32.

let's change the `countdown` argument in BunObject.zig's `Timer.setTimeout` to simply take in an `i32`, and then we can use this logic for rounding here.

Reply:
should we change `setInterval` and `sleepSync`, too?

Reply:
that makes sense to do as well, not sure if it is strictly necessary to change. do you think this rounding behavior should apply for `setInterval` and `sleepSync`?

Reply:
> that makes sense to do as well, not sure if it is strictly necessary to change. do you think this rounding behavior should apply for `setInterval` and `sleepSync`?

I need to check the specs as to whether correct behavior is allowed here, but I think they should round to the nanosecond (and toward Infinity, but that doesn't matter on real hardware), as should `setTimeout`.

Bun.sleep(Date) is a special case because we're actually hitting the limits of f64 arithmetic, though, so let's fix that one first?